x/erc20/keeper/ibc_callbacks.go

```diff
@@ -82,34 +80,45 @@ func (k Keeper) OnRecvPacket(
 	)
 
 	// check if the coin is a native staking token
-	bondDenom := k.stakingKeeper.BondDenom(ctx)
-	if coin.Denom == bondDenom {
+	if coin.Denom == evmParams.EvmDenom {
 		// no-op, received coin is the staking denomination
 		return ack
 	}
 
 	pairID := k.GetTokenPairID(ctx, coin.Denom)
-	if len(pairID) == 0 {
-		// short-circuit: if the denom is not registered, conversion will fail
-		// so we can continue with the rest of the stack
-		return ack
-	}
+	pair, found := k.GetTokenPair(ctx, pairID)
+	switch {
```

Comment:
Should we add telemetry on any of this cases?

Reply:
good idea. What metrics would you add? cc @GAtom22